src/controllers/auth.controller.js

```diff
@@ -0,0 +1,370 @@
+'use strict';
+
+const { ApiError } = require('../exceptions/api.error.js');
+const {
+  validatePassword,
+  validateEmail,
+} = require('../utils/validationData.js');
+const { userService } = require('../services/user.service.js');
+const { jwtService } = require('../services/jwt.service.js');
+const { tokenService } = require('../services/token.service.js');
+const bcrypt = require('bcrypt');
+
+const checkUserExistence = (user, id) => {
+  if (!user) {
+    throw ApiError.notFound(`There's no such user with id ${id}`);
+  }
+};
+
+const successfulMessage = (user, updatedValueName) => {
+  return {
+    message: `Your ${updatedValueName} has been successfully updated!`,
+    updatedUser: user,
+  };
+};
+
+const register = async(req, res) => {
+  const { name, email, password } = req.body;
+
+  if (!name || !email || !password) {
+    throw ApiError.badRequest('Name, email and password are required!');
+  }
+
+  const errors = {
+    password: validatePassword(password),
+    email: validateEmail(email),
+  };
+
+  if (errors.password || errors.email) {
+    throw ApiError.badRequest('Validation error!', errors);
+  }
+
+  await userService.register(name, email, password);
+
+  res.send({ message: "You've been successfully registered!" });
+};
+
+const activate = async(req, res) => {
+  const { activationToken } = req.params;
+
+  const foundUser = await userService.getUserByActivationToken(activationToken);
+
+  if (!foundUser) {
+    throw ApiError.notFound('User was not found!');
+  }
+
+  foundUser.activationToken = null;
+
+  await foundUser.save();
+
+  await sendAuthentication(res, foundUser);
+
+  res.send(foundUser);
+};
+
+const login = async(req, res) => {
+  const { email, password } = req.body;
+
+  if (!email || !password) {
+    throw ApiError.badRequest('Bad request');
+  }
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Validation error', errors);
+  }
+
+  const foundUser = await userService.getUserByEmail(email);
+
+  if (!foundUser) {
+    throw ApiError.badRequest('User with this email does not exist!');
+  }
+
+  const passwordsMatch = await bcrypt.compare(password, foundUser.password);
```

Comment:
async function move to `try/catch`

Reply:
But use middleware called catchError which already wraps the whole function in try/catch block. Or I should wrap it one more time anyway?